setup.py

```diff
@@ -1,31 +1,98 @@
+"""
+jupyter-server-proxy setup
+"""
+import json
+from pathlib import Path
+
 import setuptools
-from glob import glob
-
-with open("README.md") as f:
-    readme = f.read()
-
-setuptools.setup(
-    name="jupyter-server-proxy",
-    version='1.6.0',
-    url="https://github.com/jupyterhub/jupyter-server-proxy",
-    author="Ryan Lovett & Yuvi Panda",
-    author_email="rylo@berkeley.edu",
-    license="BSD 3-Clause",
-    description="Jupyter server extension to supervise and proxy web services",
-    long_description=readme,
+from jupyter_packaging import (
+    combine_commands,
+    create_cmdclass,
+    ensure_targets,
+    install_npm,
+    skip_if_exists,
+)
+
+HERE = Path(__file__).parent.resolve()
+
+# The name of the project
+name = "jupyter_server_proxy"
+
+lab_path = HERE / name / "labextension"
+
+# Representative files that should exist after a successful build
+jstargets = [
+    str(lab_path / "package.json"),
+]
+
+package_data_spec = {
+    name: ["*"],
+}
+
+labext_name = "jupyter_server_proxy"
+
+data_files_spec = [
+    ("share/jupyter/labextensions/%s" % labext_name, str(lab_path), "**"),
+    ("share/jupyter/labextensions/%s" % labext_name, str(HERE), "install.json"),
+    (
+        "etc/jupyter/jupyter_server_config.d",
+        "jupyter_server_proxy/etc",
+        "jupyter-server-proxy-jupyterserverextension.json",
+    ),
+    (
+        "etc/jupyter/jupyter_notebook_config.d",
+        "jupyter_server_proxy/etc",
+        "jupyter-server-proxy-notebookserverextension.json",
+    ),
+    (
+        "etc/jupyter/nbconfig/tree.d",
+        "jupyter_server_proxy/etc",
+        "jupyter-server-proxy-nbextension.json",
+    ),
+]
+
+cmdclass = create_cmdclass(
+    "jsdeps", package_data_spec=package_data_spec, data_files_spec=data_files_spec
+)
+
+js_command = combine_commands(
+    install_npm(HERE / "jupyterlab-server-proxy", build_cmd="build:prod", npm=["jlpm"]),
+    ensure_targets(jstargets),
+)
+
+is_repo = (HERE / ".git").exists()
+if is_repo:
+    cmdclass["jsdeps"] = js_command
+else:
+    cmdclass["jsdeps"] = skip_if_exists(jstargets, js_command)
+
+long_description = (HERE / "README.md").read_text()
+
+# Get the package info from package.json
+pkg_json = json.loads((HERE / "jupyterlab-server-proxy" / "package.json").read_bytes())
+
+setup_args = dict(
+    name=name,
+    version=pkg_json["version"],
```

Comment:
+1 for a major version bump, and we can make it 3 or 2. I think I prefer bumping to version 2. The motivation relates that the python package is hiding the extension package version now and I don't think there will be any conflicts.

Reply:
Maybe relevant: The [official JL extension docs](https://jupyterlab.readthedocs.io/en/stable/extension/extension_tutorial.html#publishing-your-extension) recommend to still publish the extension as a JavaScript package to npm, even after having it packaged on pip.

I believe if you decided to keep the npm package, it would make sense to bump to v3.

Reply:
3.0.0 then! :)

Reply:
Done in https://github.com/jupyterhub/jupyter-server-proxy/pull/245/commits/b7fd4b3284e963080dc45c2918aed07b003367b8.